### DISCUSSION TITLE
How do I run multiple cockpit instances on Ubuntu Server?

### DISCUSSION THREAD

**Discussion by ghost** [cockpit-project/cockpit#19096] [Q&A]
I have 30 ubuntu servers, and Cockpit doesn't perform well if I manage 10+ servers.

Now, my idea is to split the 30 servers to 10, 10, 10 groups and run multiple cockpit instance on the same server (obviously with different port) and manage these server groups. Please let me know if this is achievable 

Reference image of what I am trying to achieve
 
![image](https://github.com/cockpit-project/cockpit/assets/139256487/b972e7df-dce7-4937-90d8-d6f5137e61ef)

**Comment by martinpitt** [comment]
What exactly do you mean with "Cockpit doesn't perform well if I manage 10+ servers"? Did you add them as "remote hosts" and are connected to them all the time? What exactly becomes slow/uses a lot CPU there, the cockpit-ws process (i.e. the webserver) or something else?

The direct answer to your question is "yes" -- you can run several [cockpit/ws container instances](https://quay.io/repository/cockpit/ws) on different ports (you don't need a "cockpit vm" around them). But we really don't recommend staying connected to many machines all the time, that's not what Cockpit is designed to do.

If your desktop/laptop runs Linux, please also consider the [Cockpit Client flatpak](https://github.com/flathub/org.cockpit_project.CockpitClient), then you don't need a web server anywhere and can just connect to any of your servers via SSH (one at a time).
